common4j/src/main/com/microsoft/identity/common/java/commands/parameters/IHasExtraTokenBodyParameters.java

```diff
@@ -0,0 +1,41 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.commands.parameters;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Interface indicating that a class carries a set of string/string parameters that are intended
+ * to be merged into the OAuth2 token request body only.
+ */
+public interface IHasExtraTokenBodyParameters {
```

Comment:
>IHasExtraTokenBodyParameters [](http://example.com/codeflow?start=17&length=28)

Can we define this directly in TokenCommandParameters instead of generic interface like this?

There's already IHasExtraParameters, which is implemented by TokenRequest, TokenResponse, AuthorizationResponse and has corresponding meaning (token request body, token response body, authorization response from redirecturi). So can then be extended to be used with TokenCommandParameters.

But for stronger association (parameter to extra token parameters), defining a field in TokenCommandParameters seems better.